http/src/main/resources/reference.conf

```diff
@@ -6,6 +6,8 @@
 application {
   applicationName = "leonardo"
   #leoGoogleProject = "broad-dsde-dev"
+  # Defining leoServiceAccountJsonFile will run Leonardo in GCP mode; otherwise, it will run in Azure mode.
```

Comment:
Wondering if it's worth an explicit `application.mode = Azure` config 🤔

Reply:
maybe..the thing I want to avoid is having 
```
mode = Azure/GCP
leoServiceAccountJsonFile=<path>
```
So now we'll need to validate the 4 permutations, and reject the 2 invalid cases